### PR TITLE
Concat estimator with pigsty extensions for ConcatWith, AsVector

### PR DESCRIPTION
**Pull request by TomFinley**
This is a version of the ConcatEstimator. It is functional but will require future work in future especially w.r.t. its transformer. Related to #632.

### CODE REVIEW

File: src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs [hunk 1]
```diff
@@ -0,0 +1,424 @@
...
+            var result = inputSchema.Columns.ToDictionary(x => x.Name);
+            result[_name] = CheckInputsAndMakeColumn(inputSchema, _name, _source);
+            return new SchemaShape(result.Values);
+        }
+    }
+
+    // REVIEW: Note that the presence of this thing is a temporary measure only.
+    // If it is cleaned up by code complete so much the better, but if not we will
+    // have to wait a little bit.
+    internal sealed class ConcatTransformer : ITransformer, ICanSaveModel
```
Comment (TomFinley) [inline src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs]: This is not intended to be permanent, but is necessary for reasons of expediency. 😛 @GalOshri , @Zruty0

Comment (Ivanidzo4ka) [reply src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs]: Can we have issue, so this wouldn't be buried inside code? 

---
In reply to: [216526746](https://github.com/dotnet/machinelearning/pull/881#discussion_r216526746) [](ancestors = 216526746)

Comment (Zruty0) [reply src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs]: I'm already working on concat transformer

---
In reply to: [216746880](https://github.com/dotnet/machinelearning/pull/881#discussion_r216746880) [](ancestors = 216746880,216526746)

Comment (TomFinley) [reply src/Microsoft.ML.Data/Transforms/ConcatEstimator.cs]: I can confirm this, @Zruty0 will clean up my mess. 😃

**Pull request merged**